File: examples/python/01_basic_messaging.py
"""
01_basic_messaging.py — Send a message and read it from the inbox.

Prerequisites:
    pip install "moltmesh[grpc]"
    moltmesh-daemon &          # agent A (default socket)
    A2A_GRPC_ADDR=unix://$HOME/.moltmesh/agent_b.sock moltmesh-daemon &   # agent B

Run:
    python 01_basic_messaging.py
"""
import os
import sys
from moltmesh import A2AClient

AGENT_A_ADDR = os.getenv("AGENT_A_ADDR", "")   # default socket
AGENT_B_ADDR = os.getenv("AGENT_B_ADDR", f"unix://{os.path.expanduser('~')}/.moltmesh/agent_b.sock")

def main():
    a = A2AClient(AGENT_A_ADDR).connect()
    b = A2AClient(AGENT_B_ADDR).connect()

    id_b = b.get_identity()
    print(f"Agent B DID: {id_b.did}")

    result = a.send_message(id_b.did, text="Hello from agent A!")
    print(f"Sent message ID: {result.message_id}  queued={result.queued}")

    msgs = b.get_inbox(limit=5)
    print(f"\nAgent B inbox ({len(msgs)} message(s)):")
    for m in msgs:
        print(f"  [{m.id[:8]}] from={m.from_did}  kind={m.kind}")

    a.close()
    b.close()

if __name__ == "__main__":
    main()

File: examples/typescript/01_basic_messaging.ts
/**
 * 01_basic_messaging.ts — Send a message and read the inbox.
 *
 * Two daemons required (or use one with loopback):
 *   moltmesh-daemon &
 *   A2A_GRPC_ADDR=unix://$HOME/.moltmesh/agent_b.sock moltmesh-daemon &
 *
 * Run:
 *   npx tsx 01_basic_messaging.ts
 */
import { A2AClient } from "../../sdk/typescript/openclaw-plugin/src/client.js";

const AGENT_A_ADDR = process.env["AGENT_A_ADDR"] ?? "";
const AGENT_B_ADDR = process.env["AGENT_B_ADDR"] ?? `unix://${process.env["HOME"]}/.moltmesh/agent_b.sock`;

async function main() {
  const a = new A2AClient(AGENT_A_ADDR);
  const b = new A2AClient(AGENT_B_ADDR);

  const idB = await b.getIdentity();
  console.log(`Agent B DID: ${idB.did}`);

  const result = await a.sendMessage(idB.did, "Hello from agent A!");
  console.log(`Sent message ID: ${(result as Record<string,unknown>)["messageId"]}  queued=${(result as Record<string,unknown>)["queued"]}`);

  const msgs = await b.getInbox({ limit: 5 });
  console.log(`\nAgent B inbox (${msgs.length} message(s)):`);
  for (const m of msgs as Record<string, unknown>[]) {
    console.log(`  [${String(m["id"]).slice(0, 8)}] from=${m["fromDid"]}  kind=${m["kind"]}`);
  }

  a.close();
  b.close();
}

main().catch(console.error);
